src/main/scala/org/broadinstitute/dsde/firecloud/utils/StandardUserInfoDirectives.scala

```diff
@@ -1,46 +1,45 @@
-import spray.http.{HttpHeader, OAuth2BearerToken}
-import spray.routing.Directive1
-import spray.routing.Directives._
 
-import scala.concurrent.ExecutionContext
+import scala.concurrent.{ExecutionContext, Future}
 
-/**
- *
- * Copied wholesale from rawls on 15-Oct-2015, commit a9664c9f08d0681d6647e6611fd0c785aa8aa24a
- *
- * modified to also retrieve OIDC_CLAIM_sub/OIDC_CLAIM_user_id.
- * I could have removed the OIDC_access_token, OIDC_CLAIM_expires_in, and OIDC_CLAIM_email because we don't
- * use those in orchestration. However, they're quite lightweight, and I've left them in to keep diffs between
- * orchestration and rawls as clean as possible.
- *
- */
 trait StandardUserInfoDirectives extends UserInfoDirectives {
+  implicit val executionContext: ExecutionContext
+  val samDAO: SamDAO
 
-  def requireUserInfo(magnet: ImplicitMagnet[ExecutionContext]): Directive1[UserInfo] = {
-    implicit val ec = magnet.value
-    for(accessToken <- accessTokenHeaderDirective;
-        userEmail <- emailHeaderDirective;
-        accessTokenExpiresIn <- accessTokenExpiresInHeaderDirective;
-        id <- idHeaderDirective
-    ) yield UserInfo(userEmail, OAuth2BearerToken(accessToken), accessTokenExpiresIn.toLong, id)
-  }
+  val serviceAccountDomain = "\\S+@\\S+\\.iam\\.gserviceaccount\\.com".r
 
-  private def accessTokenHeaderDirective: Directive1[String] = headerValueByName("OIDC_access_token")
-  private def accessTokenExpiresInHeaderDirective: Directive1[String] = headerValueByName("OIDC_CLAIM_expires_in")
-  private def emailHeaderDirective: Directive1[String] = headerValueByName("OIDC_CLAIM_email")
-  private def idHeaderDirective: Directive1[String] = headerValue(extractUniqueId)
+  private def isServiceAccount(email: String) = {
+    serviceAccountDomain.pattern.matcher(email).matches
+  }
 
-  def extractUniqueId: HttpHeader => Option[String] = {hdr:HttpHeader =>
-    hdr.name match {
-      case "OIDC_CLAIM_sub" => Some(hdr.value)
-      case "OIDC_CLAIM_user_id" => Some(hdr.value)
-      case _ => None
+  override def requireUserInfo: Directive1[UserInfo] = (
+    headerValueByName("OIDC_access_token") &
+      headerValueByName("OIDC_CLAIM_user_id") &
+      headerValueByName("OIDC_CLAIM_expires_in") &
+      headerValueByName("OIDC_CLAIM_email")
+    ) tflatMap {
+    case (token, userId, expiresIn, email) => {
+      val userInfo = UserInfo(email, OAuth2BearerToken(token), expiresIn.toLong, userId)
+      onSuccess(getWorkbenchUserEmailId(userInfo).map {
+        case Some(petOwnerUser) => UserInfo(petOwnerUser.userEmail, OAuth2BearerToken(token), expiresIn.toLong, petOwnerUser.userSubjectId)
+        case None => userInfo
+      })
     }
   }
 
-
+  private def getWorkbenchUserEmailId(userInfo:UserInfo):Future[Option[RegistrationInfoV2]] = {
+    if (isServiceAccount(userInfo.userEmail)) {
+      samDAO.getRegistrationStatusV2(userInfo)
+    }
+    else {
+      Future.successful(None)
+    }
+  }
```

Comment:
this is a good change to make ... but it is also a functional change. I say we keep it in but look out for any subtle behavior changes.

Reply:
i ended up scrapping it for now. it was having a ripple effect across some other classes that used the trait. i might have screwed something up with that but i propose doing this in a followup PR instead